---
title: BD2HAN Rat ARACNe / msViper
knit: (function(inputFile, encoding) {
      out_dir <- "output"
      rmarkdown::render(inputFile,
                        encoding=encoding,
                        output_file=paste0(out_dir, "/msViper.html") })
output:
  html_document: default
---

Data is generated by Nextflow 'aracne-ap_viper' pipeline process 'viper_set'.

```{r message = FALSE, warning = FALSE}
load("../data/ARACNe/BD2HAN_2.parse_inputs.RData")
tpms <- readr::read_tsv("../data/ARACNe/BD2HAN_2_RNAseqR.exprmat.tpm.tsv")

outdir <- "output/msViper_run"
dir.create(outdir, showWarnings = FALSE, recursive = TRUE)

```

Following this, data is run through msViper functions in the background.\n
We want to test `N_LG` vs. `N_STD` and `T_STD`, i.e. the normal continuum.
Negative NES then indicates reduced expression in N_LG, and positive NES indicates
increased expression in N_LG.

```{r include = FALSE, message = FALSE, warning = FALSE, comment = NA}
library(tidyverse)
library(Biobase)

TAG <- "BD2HAN_2"
LEVELS <- levels(factor(phenos$description))
pairname <- "N_LG-vs-STD"

signature <- viper::rowTtest(x = combined.eset,
                             pheno = "description",
                             group1 = "N_LG",
                             group2 = c("N_STD", "T_STD"))
rownamesp <- rownames(signature$p.value)
signature$p.value <- as.matrix(p.adjust(signature$p.value, method="BH"))
rownames(signature$p.value) <- rownamesp
signature <- (qnorm(signature$p.value/2, lower.tail = FALSE) * sign(signature$statistic))[, 1]
nullmodel <- viper::ttestNull(x = combined.eset,
                              pheno = "description",
                              group1 = "N_LG",
                              group2 = c("N_STD", "T_STD"),
                              per = 5000,
                              repos = TRUE,
                              verbose = TRUE)

## msViper, leading edge analyses + bootstrapped
mra <- viper::msviper(ges = signature,
                      regulon = regulonaracne,
                      nullmodel = nullmodel,
                      verbose = FALSE)

mra <- viper::ledge(mra)
mra <- viper::shadow(mobj = mra,
                     regulators = 0.01,
                     shadow = 0.01,
                     per = 5000,
                     verbose = FALSE)

mrac <- viper::msviperCombinatorial(mobj = mra,
                                    regulators = 0.01,
                                    verbose = FALSE)

##if statement to deny error from synergy analysis which otherwise exits from the script
if(length(mrac$regulon) > length(mra$regulon)){
  mra <- viper::msviperSynergy(mobj = mrac, verbose = FALSE)
  ##BH adjust p-values
  mra$es$q.value <- p.adjust(mra$es$p.value, method = "BH")
}
if(length(mrac$regulon) == length(mra$regulon)){
  ##BH adjust p-values
  mra$es$q.value <- p.adjust(mra$es$p.value, method = "BH")
}
mra$signature <- na.omit(mra$signature)

mra_tab <- data.frame(NES = mra$es$nes,
                      Size = mra$es$size,
                      p.value = mra$es$p.value,
                      q.value = mra$es$q.value)

save(signature, nullmodel, mra, mra_tab, file = paste0(outdir, "/", TAG, ".", pairname, ".RData"))
readr::write_tsv(mra_tab, file = paste0(outdir, "/", TAG, ".", pairname, ".msViper.results.tsv"))
```

******
\newpage

## Plots and Results
MRA plot and table of p<0.1 genes annotated with gene names

Gene expression plots of genes found significant from MRA table
```{r echo = FALSE, message = FALSE, warning = FALSE, fig.width = 12, fig.height = 10, comment = NA}
plot(mra, pval = mra$es$q.value, cex = .7)

mra_sig <- tibble::as_tibble(mra_tab, rownames = "ensembl_gene_id") %>%
           dplyr::filter(p.value < 0.1) %>%
           dplyr::right_join(tx2gene[,c("external_gene_name", "ensembl_gene_id")], .) %>%
           dplyr::distinct()
DT::datatable(data = mra_sig,
              extensions = 'Buttons',
              options = list(dom = 'Bfrtip',buttons = 'excel'))
mra_sig_genes <- dplyr::filter(.data = tpms,
                              ensembl_gene_id %in% mra_sig$ensembl_gene_id) %>%
                dplyr::left_join(., tx2gene[c("ensembl_gene_id", "external_gene_name")]) %>%
                dplyr::select(external_gene_name, everything(), -ensembl_gene_id) %>%
                dplyr::distinct() %>%
                dplyr::mutate(dplyr::across(where(is.numeric), log2)) %>%
                tidyr::pivot_longer(!external_gene_name,
                                    names_to = "sampleID",
                                    values_to = "log2TPM") %>%
                dplyr::left_join(., phenos)

ggp <- ggplot2::ggplot(data = mra_sig_genes,
                aes(x = description, y = log2TPM, fill = description)) +
      ggplot2::geom_violin(aes(fill = description, color = description)) +
      ggsignif::geom_signif(comparisons = list(c("N_LG","N_STD"),
                                               c("N_LG","T_STD"),
                                               c("N_STD","T_STD")),
                            map_signif_level = TRUE,
                            step_increase=0.1,
                            textsize=3.2) +
      ggforce::facet_wrap_paginate(~external_gene_name, ncol = 3, nrow = 3, scales = "free")
for(x in 1:ggforce::n_pages(ggp)){
    print(ggp +
          ggforce::facet_wrap_paginate(~external_gene_name, scales = "free",
                                       ncol = 3, nrow = 3, , page = x))
}
```

******
\newpage

## Genes regulated by Prss16 (ENSRNOG00000057865)
From Leading Edge analysis
Q: What is expression profile of those genes regulated by Prss16?
```{r echo = FALSE, message = FALSE, warning = FALSE, fig.width = 12, fig.height = 10, comment = NA}
t1 <- readr::read_tsv("../data/ARACNe/BD2HAN_2_RNAseqR.exprmat.tpm.tsv")
prss16_genes <- dplyr::filter(.data = t1,
                              ensembl_gene_id %in% c("ENSRNOG00000057865", mra$ledge$ENSRNOG00000057865)) %>%
                dplyr::left_join(., tx2gene[c("ensembl_gene_id", "external_gene_name")]) %>%
                dplyr::select(external_gene_name, everything(), -ensembl_gene_id) %>%
                dplyr::distinct() %>%
                dplyr::mutate(dplyr::across(where(is.numeric), log2)) %>%
                tidyr::pivot_longer(!external_gene_name,
                                    names_to = "sampleID",
                                    values_to = "log2TPM") %>%
                dplyr::left_join(., phenos)

ggp <- ggplot2::ggplot(data = prss16_genes,
                aes(x = description, y = log2TPM, fill = description)) +
      ggplot2::geom_violin(aes(fill = description, color = description)) +
      ggsignif::geom_signif(comparisons = list(c("N_LG","N_STD"),
                                               c("N_LG","T_STD"),
                                               c("N_STD","T_STD")),
                            map_signif_level = TRUE,
                            step_increase=0.1,
                            textsize=3.2) +
      ggforce::facet_wrap_paginate(~external_gene_name, ncol = 3, nrow = 3, scales = "free")
for(x in 1:ggforce::n_pages(ggp)){
    print(ggp +
          ggforce::facet_wrap_paginate(~external_gene_name, scales = "free",
                                       ncol = 3, nrow = 3, , page = x))
}
```

******
